### PR TITLE
AGL US 23-24

### PR DESCRIPTION
**Pull request by alex-latham**
## All Submissions:

* [X] Merge the target branch into the PR branch. Fix any conflicts that might appear.
* [X] Have you checked to ensure that other pull request are approved for merge [Pull Requests](../../../pulls)?
* [X] What stories does the PR include
  - 23 & 24
* [X] What issues does the PR address
  - Removes persistent "Logged in as" text in navbar (only shows when logged in now)

### New Feature Submissions:

* [X] Have you added an explanation of what your changes do?

* [X] Have you lint your code locally prior to submission?

### Testing:

* [X] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully ran tests with your changes locally?
* [X] All new and existing tests passed.

**Test coverage** = 100%
--------------------------------------------------------------------------------
### Change Log
  - Adds increment/decrement functionality to increase/decrease quantity of cart items (lower bound 0 removes item from cart, upper bound at item.inventory)
  - Also adds cart model testing
  - Add .DS_Store to ignore file

### CODE REVIEW

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -27,14 +31,14 @@ def remove_item
...
-  # def increment_decrement
-  #   if params[:increment_decrement] == "increment"
-  #     cart.add_quantity(params[:item_id]) unless cart.limit_reached?(params[:item_id])
-  #   elsif params[:increment_decrement] == "decrement"
-  #     cart.subtract_quantity(params[:item_id])
-  #     return remove_item if cart.quantity_zero?(params[:item_id])
-  #   end
-  #   redirect_to "/cart"
-  # end
+
```
Comment (tylertomlinson) [inline app/controllers/cart_controller.rb:34]: Interesting how this played out. Was wondering why it was commented out.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -21,7 +21,7 @@
       <%= link_to "Dashboard", "/merchant" if current_merchant?%>
       <%= link_to "Dashboard", "/admin" if current_admin?%>
       <%= link_to "Users", "/admin/users" if current_admin?%>
-      <p>Logged in as <%=current_user.name if current_user%> </p>
+      <%= "Logged in as #{current_user.name}" if current_user%>
```
Comment (tylertomlinson) [inline app/views/layouts/application.html.erb:24]: Will this take care of "logged in as" displaying on nav bar regardless?

File: spec/models/cart_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
```
Comment (tylertomlinson) [inline spec/models/cart_spec.rb:1]: Solid model test

File: .gitignore [hunk 4]
```diff
@@ -18,3 +18,4 @@ coverage
 /yarn-error.log
 
 .byebug_history
+.DS_Store
```
Comment (tylertomlinson) [inline .gitignore:21]: Does everyone not have a global git_ignore?

Comment (tylertomlinson) [reply .gitignore:21]: if not then coverage should live here too

**Review by danielpselph — APPROVED** [review]
Great work on the increment decrement method

**Review by tylertomlinson — APPROVED** [review]

**Pull request merged**